### PR TITLE
Init package with CameraService v1

### PR DESCRIPTION
**Pull request by FejZa**
# Reality Collective - Reality Toolkit Pull Request

## Overview

This is a carry over from core into a dedicated package. Here's the overview of important changes compared to the previous camera service implementation:

- Renamed `MixedRealityCameraSystem` to `CameraService`. All related scripts accordingly
- The camera rig is now a prefab configured in the platforms `CameraServiceModule`'s profile
- The service checks whether a valid rig exists in the scene when it's added to the service configuration profile and initializes it if not. It logs warnings or errors if something is not as needed by the service to work
- The service will ensure components, such as the `TrackedPoseDriver` are added to the rig, if the prefab does not have them

TLDR: The rig is now a prefab

### CODE REVIEW

**Comment by SimonDarksideJ** [comment]
When importing without the toolkit present, I get the following errors:
image

**Comment by SimonDarksideJ** [comment]
Maybe we should move the package installer to Utilities?

**Comment by SimonDarksideJ** [comment]
No property listed for the new prefab variable in the editor
image
Likely needs the inspector checking

**Comment by SimonDarksideJ** [comment]
Removing the Camera service DELETES the existing rig in the scene, don't think it should do that.   Shouldn't automatically clear up when the service is removed

**Comment by SimonDarksideJ** [comment]
If I alter the XRCameraRig in the scene, my changes are destroyed when the camera service is viewed in the Service Manager inspector.  I.E. I add some components or additional rig objects.

What is the expected behaviour here?

**Comment by SimonDarksideJ** [comment]
Build script does not have all the required dependencies, 

```yml
      dependencies: '{"development": "github.com/realitycollective/com.realitycollective.buildtools.git"}'
    secrets: inherit
```

Missing the Service Framework for the test build

**Comment by FejZa** [comment]
Thanks @SimonDarksideJ, I'll look into the clean up and inspector. Adding the package without the toolkit is not a supported scenario and hence not an issue.

**Comment by FejZa** [comment]
Well it's a PREFAB, you are not supposed to edit it in the scene and not apply the changes

**Comment by FejZa** [comment]
But let me look into a way to maybe support this or at leat auto-apply changes. Might not even be an issue anymore once the service does not delete the rig on clean up

**Comment by SimonDarksideJ** [comment]
Prefab is being destroyed and recreated in the scene whenever the camera service is inspected.

Replication Instructions:
1. Add the camera service and set the runtime platforms (everything used for testing)
2. See the XRCameraRig prefab loaded into the scene
3. Rename the XRCamera prefab
4. Inspect the camera system again

Expected:
The Rig and managing script are still in the scene, the prefab should remain untouched

Actual:
The rig is destroyed and replaced with the original

**Comment by SimonDarksideJ** [comment]
The scene is not marked dirty when the camera is added to the scene:

Replication Instructions:
1. Create a new scene and add the service manager
2. Add the camera service and set the runtime platforms (everything used for testing)
3. See the XRCameraRig prefab loaded into the scene and scene is dirty.
4. Remove the XRCameraRig and save (scene now has Service Manager and Light ONLY)
5. Open the scene file in a text editor for comparison
6. Open the scene again in Unity
7. Note that the XRCameraRig is added but scene is not marked as dirty
8. Open the unity scene file again and compare, no change found
9. Return to Unity, Save the scene and reopen
10. Compare scene file again, no change

Expected:
If and when the scene is altered by the Camera Service, it should be marked as dirty

Actual:
Apart from the initial change when the Camera is removed and the first prefab added, the scene is never marked dirty.

**Comment by SimonDarksideJ** [comment]
> Well it's a PREFAB, you are not supposed to edit it in the scene and not apply the changes

Fair enough, we should ensure this is clearly documented that each camera provider will give its own prefab configured in the way it should be used.  The flow being that the XRRig will be replaced at runtime with the prefab required.

**Review by SimonDarksideJ — APPROVED** [review]
A fantastic start
